New clickhouse schema (#137)
flant/loghouse: README.md

# Installation

> To install loghouse, you need to have [Helm](https://github.com/kubernetes/helm) in your Kubernetes cluster. Also, our chart expecting that you install [cert-manager](https://github.com/jetstack/cert-manager) to your cluser. 

Reviewer: А helm3 подойдёт же? "in your cluster" это ведь для второго хельма уточнение, а теперь похоже лишнее...

```suggestion
To install loghouse, you need to use [Helm](https://github.com/kubernetes/helm). Also, it is considered that [cert-manager](https://github.com/jetstack/cert-manager) is already installed in your cluster. 
```

**Note on logs format**: If log entry is in JSON, it will be formatted according to its values' types, i.e. each field will be stored in corresponding table: string_fields, number_fields, boolean_fields, null_fields or labels (the last one is used for containers labels to make further filtering and lookups easy). ClickHouse built-in functions will be used to process these data types. If log entry isn't in JSON, it will be stored in string_fields table.

> Currently, ClickHouse DBMS is deployed as a single instance via Deployment which brings this instance to a random K8s node (this behaviour can be changed by using nodeSelector and tolerations to choose a specific node). ClickHouse stores its data in hostPath volume or Persistent Volumes Claim (PVC) created with any storageClass you prefer. You can find extended schemas for clickhouse [here](docs/schemas/README.md)

Reviewer: В русской доке написано StatefulSet :thinking:

Web UI ([screenshot](docs/loghouse_interface.png)) is composed of two components:

* **frontend** — nginx with basic authorization. This authorization is used to limit user's access with logs from given Kubernetes namespaces only;
* **backend** — Ruby application displaying logs from ClickHouse.

# Upgrading

> Helm chart was rewrited. All kubernetes API for objects was updated. Helm hook policy was updated too. So, for update to **0.3** you need to remove some conflicting objects.

Reviewer: ```suggestion
Helm chart was rewritten. All API versions were updated. Helm hook policy was updated too. So, to make an update to **0.3** you need to remove some conflicting objects.
```

```
kubectl -n loghouse delete jobs,ing --all
```

> **Warning!** Database schema was changed. Create backup before upgrade. Migration task will be started after upgrade.

Reviewer: ```suggestion
**Warning!** Database schema was changed. Please, create a backup of clickhouse data before upgrade. Note that the migration task will automatically start at the end of the upgrade process.
```
